src/features/customerForm/CustomerEditForm.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useMutation, useQuery } from '@apollo/react-hooks';
+
+import CustomerForm from './CustomerForm';
+import { CustomerFormValues, FormProps } from './types';
+import LoadingSpinner from '../../common/spinner/LoadingSpinner';
+import { CUSTOMER_FORM_QUERY } from './queries';
+import { CUSTOMER_FORM } from './__generated__/CUSTOMER_FORM';
+import { createUpdateInputs, getCustomerFormValues, getIdentifiers } from './utils';
+import { UPDATE_BERTH_SERVICES_PROFILE_MUTATION, UPDATE_PROFILE_MUTATION } from './mutations';
+import { UPDATE_PROFILE, UPDATE_PROFILEVariables as UPDATE_PROFILE_VARS } from './__generated__/UPDATE_PROFILE';
+import {
+  UPDATE_BERTH_SERVICES_PROFILE,
+  UPDATE_BERTH_SERVICES_PROFILEVariables as UPDATE_BERTH_SERVICES_PROFILE_VARS,
+} from './__generated__/UPDATE_BERTH_SERVICES_PROFILE';
+
+export interface CustomerEditFormProps
+  extends Omit<FormProps<CustomerFormValues>, 'initialValues' | 'onCreate' | 'onDelete'> {
+  customerId: string;
+}
+
+const CustomerEditForm = ({ customerId, onCancel, onSubmit, refetchQueries }: CustomerEditFormProps) => {
+  const { t } = useTranslation();
+
+  const { loading, error, data } = useQuery<CUSTOMER_FORM>(CUSTOMER_FORM_QUERY, {
+    variables: { id: customerId },
+  });
+
+  const [updateProfile, { loading: updateProfileLoading }] = useMutation<UPDATE_PROFILE, UPDATE_PROFILE_VARS>(
+    UPDATE_PROFILE_MUTATION
+  );
+  const [updateBerthServicesProfile, { loading: updateBerthServicesLoading }] = useMutation<
+    UPDATE_BERTH_SERVICES_PROFILE,
+    UPDATE_BERTH_SERVICES_PROFILE_VARS
+  >(UPDATE_BERTH_SERVICES_PROFILE_MUTATION, {
+    refetchQueries: refetchQueries ?? [],
+  });
+
+  if (loading) return <LoadingSpinner isLoading={loading} />;
+  if (error || !data?.profile) return <div>{t('forms.common.error')}</div>;
```

Comment:
I think we should let our global error handler to kick in in case of error, so should this be:
`if (!data?.profile)` ?

Reply:
Sure, that could be better. What about the other forms that use this pattern?

Reply:
I think we should clean those places also, you can do it in this PR if you want.

Reply:
There are actually several components that use this `if (error...` pattern. I think it would be better to separate it into a new ticket, as the changes to error handling should be properly tested.